sklearn/feature_selection/from_model.py

```diff
@@ -111,3 +133,93 @@ def transform(self, X, threshold=None):
             return X[:, mask]
         else:
             raise ValueError("Invalid threshold: all features are discarded.")
+
+
+class SelectFromModel(BaseEstimator, SelectorMixin):
+    """Meta-transformer for selecting features based on importance
+       weights.
+
+    Parameters
+    ----------
+    estimator : object
+        The base estimator from which the transformer is built.
+
+    threshold : string, float, optional
+        The threshold value to use for feature selection. Features whose
+        importance is greater or equal are kept while the others are
+        discarded. If "median" (resp. "mean"), then the threshold value is
+        the median (resp. the mean) of the feature importances. A scaling
+        factor (e.g., "1.25*mean") may also be used. If None and if
+        available, the object attribute ``threshold`` is used. Otherwise,
+        "mean" is used by default.
+
+    Attributes
+    ----------
+    `estimator_`: an estimator
+        The base estimator from which the transformer is built.
+
+    `scores_`: array, shape(n_features,)
+        The importance of each feature according to the fit model.
+
+    `threshold_`: float
+        The threshold value used for feature selection.
+    """
+
+    def __init__(self, estimator, threshold=None):
+        self.estimator = estimator
+        self.threshold = threshold
+
+    def _get_support_mask(self):
+        self.threshold_ = _calculate_threshold(self.estimator, self.scores_,
+                                               self.threshold)
+        return self.scores_ >= self.threshold_
```

Comment:
In terms of implementation it would be fairly easy to do: it boils down to replace occurences of `self.scores_` by ` _get_feature_importances(self.estimator_, X)`.